### PR TITLE
Bump to Expo 52 and RN 76

### PR DESCRIPTION
**Pull request by renanmav**
Enables react-native new architecture.

Some links:
- https://reactnative.dev/blog/2024/10/23/release-0.76-new-architecture
- https://reactnative.dev/blog/2024/10/23/the-new-architecture-is-here
- https://expo.dev/changelog/2024/10-24-sdk-52-beta

Checklist:
- [ ] Run expo-doctor

### CODE REVIEW

**Comment by renanmav** [comment]
Getting this error on metro bundler

```
iOS Bundled 18297ms node_modules/expo-router/entry.js (3745 modules)
 (NOBRIDGE) LOG  Bridgeless mode is enabled
 INFO
 💡 JavaScript logs will be removed from Metro in React Native 0.77! Please use React Native DevTools as your default tool. Tip: Type j in the terminal to open (requires Google Chrome or Microsoft Edge).
 (NOBRIDGE) ERROR  Warning: Error: Couldn't register the navigator. Have you wrapped your app with 'NavigationContainer'?

This can also happen if there are multiple copies of '@react-navigation' packages installed.

This error is located at:
    in NativeStackNavigator
    in Unknown (created by RootLayout)
    in ThemeProvider (created by RootLayout)
    in PersistGate (created by RootLayout)
    in Provider (created by RootLayout)
    in RCTView (created by View)
    in View (created by GestureHandlerRootView)
    in GestureHandlerRootView (created by RootLayout)
    in RootLayout
    in Unknown (created by Route())
    in Suspense (created by Route())
    in Route (created by Route())
    in Route() (created by ContextNavigator)
    in RNCSafeAreaProvider (created by SafeAreaProvider)
    in SafeAreaProvider (created by wrapper)
    in wrapper (created by ContextNavigator)
    in ThemeProvider
    in EnsureSingleNavigator
    in BaseNavigationContainer
    in NavigationContainerInner (created by ContextNavigator)
    in ContextNavigator (created by ExpoRoot)
    in ExpoRoot (created by App)
    in App (created by ErrorOverlay)
    in ErrorToastContainer (created by ErrorOverlay)
    in ErrorOverlay (created by withDevTools(ErrorOverlay))
    in withDevTools(ErrorOverlay)
    in RCTView (created by View)
    in View (created by AppContainer)
    in RCTView (created by View)
    in View (created by AppContainer)
    in AppContainer
    in main(RootComponent)
```

File: app.json
Comment (MeisterJustice) [inline app.json:1]: @renanmav curious to know why we're replacing this file with `app.config.js`. I don't see any comment on that in the expo change-logs

**Review by MeisterJustice — COMMENTED** [review]
Also, the app isn't building on my end. Already ran prebuild and installed dependencies. I'll come back to this later

File: app.json
Comment (renanmav) [reply app.json:1]: @MeisterJustice just for type intellisense.

https://github.com/metastable-labs/liquid-interface/blob/83f7eb329cb1374198171f805d7739b410770390/app.config.js#L1

But that can also be accomplished using the Expo VSCode extension + `app.json`

**Review by MeisterJustice — APPROVED** [review]

**Pull request merged**